src/components/form/date-input/date-input-component.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { DateInputComponent } from './date-input.component';
+import { DebugElement } from '@angular/core';
+import { PaletteUtilsModule } from '../../../shared/components/palette/utils/utils.module';
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import { MockComponent } from 'ng2-mock-component';
+import createSpy = jasmine.createSpy;
+
+describe("Date input component", function () {
+
+    let fixture: ComponentFixture<DateInputComponent>;
+    let component: DateInputComponent;
+    let de: DebugElement;
+
+    let Input: any;
+    let onChange;
+
+    beforeEach(async(() => {
+        // Input is mocked so that one-way bound inputs can be tested
+        Input = MockComponent({
+            selector: 'input', inputs: [
+                'type',
+                'value',
+                'change',
+                'keyup',
+                'disabled'
+            ]
+        });
+        TestBed
+            .configureTestingModule({
+                imports: [
+                    ReactiveFormsModule,
+                    PaletteUtilsModule
+                ],
+                declarations: [
+                    DateInputComponent,
+
+                    // Mock
+                    Input
+                ],
+                providers: []
+            })
+            .compileComponents();
+
+        fixture = TestBed.createComponent(DateInputComponent);
+        component = fixture.componentInstance;
+        onChange = createSpy('onChange');
+
+        component.registerOnChange(onChange);
+
+        de = fixture.debugElement;
+        fixture.detectChanges();
+    }));
+
+    it('should be valid when day value', async () => {
+        component.id="dayInput"         
+        component.dayChange('09');
+        component.displayDay = '09'
+        fixture.detectChanges();
+        let input = await de.query(By.css(`#${component.dayId()}`)).componentInstance;
+        expect(input.value).toBe('09');
+    });
+
+    it('Day input should be valid when null', async () => {
```

Comment:
Could I ask you to move the `Day Input` into a `describe` wrapper please? Same for `Month Input` and `Year Input` please.

Reply:
I have done the changes as requested. Can you please review it again?